packages/x-date-pickers/src/DesktopNextDatePicker/DesktopNextDatePicker.types.ts

```diff
@@ -27,11 +28,23 @@ export interface DesktopNextDatePickerProps<TDate>
   /**
    * Overrideable components.
    * @default {}
+   * @deprecated
    */
   components?: DesktopNextDatePickerSlotsComponent<TDate>;
   /**
    * The props used for each component slot.
    * @default {}
+   * @deprecated
    */
   componentsProps?: DesktopNextDatePickerSlotsComponentsProps<TDate>;
+  /**
+   * Overrideable components.
+   * @default {}
+   */
+  slots?: UncapitalizeObjectKeys<DesktopNextDatePickerSlotsComponent<TDate>>;
```

Comment:
Could maybe the `SlotsAndSlotsProps` utility type be used for these cases as well? 🤔 
I see it used in some places, but in others there are lots of duplication. 🤔

Reply:
It would return the following error
```
Named property 'components' of types 
'BaseNextDatePickerProps<TDate>' and 
'SlotsAndSlotsProps<DesktopNextDatePickerSlotsComponent<TDate>, DesktopNextDatePickerSlotsComponentsProps<TDate>>' 
are not identical
```

Adding `components` solve this error by overriding it. So I need to do the same with `slots`

Reply:
Gotcha. 👍

Reply:
I've tinkered with types and to me it still seems worth it to play around the interface extending issue in order to achieve WAY cleaner types for every component. 🤔 
I've tried doing it for several components with 0 issues...
What do you think?
```diff
diff --git a/packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.types.ts b/packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.types.ts
index 599879893..9535d727b 100644
--- a/packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.types.ts
+++ b/packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.types.ts
@@ -15,7 +15,7 @@ import {
   PickersCalendarHeaderSlotsComponentsProps,
   DayCalendarProps,
   ExportedUseViewsOptions,
-  UncapitalizeObjectKeys,
+  SlotsAndSlotsProps,
 } from '@mui/x-date-pickers/internals';
 import { DateRange, RangePositionProps, DayRangeValidationProps } from '../internal/models';
 import { DateRangeCalendarClasses } from './dateRangeCalendarClasses';
@@ -97,7 +97,11 @@ export interface ExportedDateRangeCalendarProps<TDate>
 
 export interface DateRangeCalendarProps<TDate>
   extends ExportedDateRangeCalendarProps<TDate>,
-    Partial<RangePositionProps> {
+    Partial<RangePositionProps>,
+    SlotsAndSlotsProps<
+      DateRangeCalendarSlotsComponent<TDate>,
+      DateRangeCalendarSlotsComponentsProps<TDate>
+    > {
   /**
    * The selected value.
    * Used when the component is controlled.
@@ -121,28 +125,6 @@ export interface DateRangeCalendarProps<TDate>
    * The system prop that allows defining system overrides as well as additional CSS styles.
    */
   sx?: SxProps<Theme>;
-  /**
-   * Overrideable components.
-   * @default {}
-   * @deprecated Please use `slots` with uncapitalized properties.
-   */
-  components?: DateRangeCalendarSlotsComponent<TDate>;
-  /**
-   * The props used for each component slot.
-   * @default {}
-   * @deprecated Please use `slotsProps`
-   */
-  componentsProps?: DateRangeCalendarSlotsComponentsProps<TDate>;
-  /**
-   * Overrideable component slots.
-   * @default {}
-   */
-  slots?: UncapitalizeObjectKeys<DateRangeCalendarSlotsComponent<TDate>>;
-  /**
-   * The props used for each component slot.
-   * @default {}
-   */
-  slotsProps?: DateRangeCalendarSlotsComponentsProps<TDate>;
 }
 
 export interface DateRangeCalendarOwnerState<TDate> extends DateRangeCalendarProps<TDate> {
diff --git a/packages/x-date-pickers/src/DesktopNextDatePicker/DesktopNextDatePicker.types.ts b/packages/x-date-pickers/src/DesktopNextDatePicker/DesktopNextDatePicker.types.ts
index 3e753295c..e501648d0 100644
--- a/packages/x-date-pickers/src/DesktopNextDatePicker/DesktopNextDatePicker.types.ts
+++ b/packages/x-date-pickers/src/DesktopNextDatePicker/DesktopNextDatePicker.types.ts
@@ -11,7 +11,7 @@ import {
 import { MakeOptional } from '../internals/models/helpers';
 import { BaseNextNonStaticPickerExternalProps } from '../internals/models/props/basePickerProps';
 import { DateView } from '../internals/models/views';
-import { UncapitalizeObjectKeys } from '../internals/utils/slots-migration';
+import { SlotsAndSlotsProps, WithoutSlots } from '../internals/utils/slots-migration';
 
 export interface DesktopNextDatePickerSlotsComponent<TDate>
   extends BaseNextDatePickerSlotsComponent<TDate>,
@@ -22,29 +22,10 @@ export interface DesktopNextDatePickerSlotsComponentsProps<TDate>
     UseDesktopPickerSlotsComponentsProps<TDate, DateView> {}
 
 export interface DesktopNextDatePickerProps<TDate>
-  extends BaseNextDatePickerProps<TDate>,
+  extends WithoutSlots<BaseNextDatePickerProps<TDate>>,
     DesktopOnlyPickerProps<TDate>,
-    BaseNextNonStaticPickerExternalProps {
-  /**
-   * Overrideable components.
-   * @default {}
-   * @deprecated Please use `slots` with uncapitalized properties.
-   */
-  components?: DesktopNextDatePickerSlotsComponent<TDate>;
-  /**
-   * The props used for each component slot.
-   * @default {}
-   * @deprecated Please use `slotsProps`
-   */
-  componentsProps?: DesktopNextDatePickerSlotsComponentsProps<TDate>;
-  /**
-   * Overrideable component slots.
-   * @default {}
-   */
-  slots?: UncapitalizeObjectKeys<DesktopNextDatePickerSlotsComponent<TDate>>;
-  /**
-   * The props used for each component slot.
-   * @default {}
-   */
-  slotsProps?: DesktopNextDatePickerSlotsComponentsProps<TDate>;
-}
+    BaseNextNonStaticPickerExternalProps,
+    SlotsAndSlotsProps<
+      DesktopNextDatePickerSlotsComponent<TDate>,
+      DesktopNextDatePickerSlotsComponentsProps<TDate>
+    > {}
diff --git a/packages/x-date-pickers/src/DesktopNextDateTimePicker/DesktopNextDateTimePicker.types.ts b/packages/x-date-pickers/src/DesktopNextDateTimePicker/DesktopNextDateTimePicker.types.ts
index e8c32ec0d..a91e553fb 100644
--- a/packages/x-date-pickers/src/DesktopNextDateTimePicker/DesktopNextDateTimePicker.types.ts
+++ b/packages/x-date-pickers/src/DesktopNextDateTimePicker/DesktopNextDateTimePicker.types.ts
@@ -11,7 +11,7 @@ import {
 import { MakeOptional } from '../internals/models/helpers';
 import { BaseNextNonStaticPickerExternalProps } from '../internals/models/props/basePickerProps';
 import { DateOrTimeView } from '../internals/models/views';
-import { UncapitalizeObjectKeys } from '../internals/utils/slots-migration';
+import { SlotsAndSlotsProps, WithoutSlots } from '../internals/utils/slots-migration';
 
 export interface DesktopNextDateTimePickerSlotsComponent<TDate>
   extends BaseNextDateTimePickerSlotsComponent<TDate>,
@@ -25,29 +25,10 @@ export interface DesktopNextDateTimePickerSlotsComponentsProps<TDate>
     UseDesktopPickerSlotsComponentsProps<TDate, DateOrTimeView> {}
 
 export interface DesktopNextDateTimePickerProps<TDate>
-  extends BaseNextDateTimePickerProps<TDate>,
+  extends WithoutSlots<BaseNextDateTimePickerProps<TDate>>,
     DesktopOnlyPickerProps<TDate>,
-    BaseNextNonStaticPickerExternalProps {
-  /**
-   * Overrideable components.
-   * @default {}
-   * @deprecated Please use `slots` with uncapitalized properties.
-   */
-  components?: DesktopNextDateTimePickerSlotsComponent<TDate>;
-  /**
-   * The props used for each component slot.
-   * @default {}
-   * @deprecated Please use `slotsProps`
-   */
-  componentsProps?: DesktopNextDateTimePickerSlotsComponentsProps<TDate>;
-  /**
-   * Overrideable component slots.
-   * @default {}
-   */
-  slots?: UncapitalizeObjectKeys<DesktopNextDateTimePickerSlotsComponent<TDate>>;
-  /**
-   * The props used for each component slot.
-   * @default {}
-   */
-  slotsProps?: DesktopNextDateTimePickerSlotsComponentsProps<TDate>;
-}
+    BaseNextNonStaticPickerExternalProps,
+    SlotsAndSlotsProps<
+      DesktopNextDateTimePickerSlotsComponent<TDate>,
+      DesktopNextDateTimePickerSlotsComponentsProps<TDate>
+    > {}
diff --git a/packages/x-date-pickers/src/DesktopNextTimePicker/DesktopNextTimePicker.types.ts b/packages/x-date-pickers/src/DesktopNextTimePicker/DesktopNextTimePicker.types.ts
index 864dfd7f0..346be7362 100644
--- a/packages/x-date-pickers/src/DesktopNextTimePicker/DesktopNextTimePicker.types.ts
+++ b/packages/x-date-pickers/src/DesktopNextTimePicker/DesktopNextTimePicker.types.ts
@@ -11,7 +11,7 @@ import {
 import { MakeOptional } from '../internals/models/helpers';
 import { BaseNextNonStaticPickerExternalProps } from '../internals/models/props/basePickerProps';
 import { TimeView } from '../internals/models/views';
-import { UncapitalizeObjectKeys } from '../internals/utils/slots-migration';
+import { SlotsAndSlotsProps, WithoutSlots } from '../internals/utils/slots-migration';
 
 export interface DesktopNextTimePickerSlotsComponent<TDate>
   extends BaseNextTimePickerSlotsComponent<TDate>,
@@ -22,29 +22,10 @@ export interface DesktopNextTimePickerSlotsComponentsProps<TDate>
     UseDesktopPickerSlotsComponentsProps<TDate, TimeView> {}
 
 export interface DesktopNextTimePickerProps<TDate>
-  extends BaseNextTimePickerProps<TDate>,
+  extends WithoutSlots<BaseNextTimePickerProps<TDate>>,
     DesktopOnlyPickerProps<TDate>,
-    BaseNextNonStaticPickerExternalProps {
-  /**
-   * Overrideable components.
-   * @default {}
-   * @deprecated Please use `slots` with uncapitalized properties.
-   */
-  components?: DesktopNextTimePickerSlotsComponent<TDate>;
-  /**
-   * The props used for each component slot.
-   * @default {}
-   * @deprecated Please use `slotsProps`
-   */
-  componentsProps?: DesktopNextTimePickerSlotsComponentsProps<TDate>;
-  /**
-   * Overrideable component slots.
-   * @default {}
-   */
-  slots?: UncapitalizeObjectKeys<DesktopNextTimePickerSlotsComponent<TDate>>;
-  /**
-   * The props used for each component slot.
-   * @default {}
-   */
-  slotsProps?: DesktopNextTimePickerSlotsComponentsProps<TDate>;
-}
+    BaseNextNonStaticPickerExternalProps,
+    SlotsAndSlotsProps<
+      DesktopNextTimePickerSlotsComponent<TDate>,
+      DesktopNextTimePickerSlotsComponentsProps<TDate>
+    > {}
diff --git a/packages/x-date-pickers/src/internals/utils/slots-migration.ts b/packages/x-date-pickers/src/internals/utils/slots-migration.ts
index 58b0c4c22..ccb8f8fc6 100644
--- a/packages/x-date-pickers/src/internals/utils/slots-migration.ts
+++ b/packages/x-date-pickers/src/internals/utils/slots-migration.ts
@@ -19,17 +19,19 @@ export type UncapitalizeObjectKeys<T extends object> = {
     : never;
 };
 
+export type WithoutSlots<T> = Omit<T, 'components' | 'componentsProps' | 'slots' | 'slotsProps'>;
+
 export interface SlotsAndSlotsProps<TSlots extends object, TSlotsProps> {
   /**
    * Overrideable components.
    * @default {}
-   * @deprecated Please use `slots` with uncapitalized properties.
+   * @deprecated Please use `slots`.
    */
   components?: TSlots;
   /**
    * The props used for each component slot.
    * @default {}
-   * @deprecated Please use `slotsProps`
+   * @deprecated Please use `slotsProps`.
    */
   componentsProps?: TSlotsProps;
   /**

```